recipes/igblast/build.sh

```diff
@@ -1,45 +1,71 @@
-    patchelf --replace-needed libbz2.so.1 libbz2.so.1.0 bin/$name
-  done
-fi
-
 # $SHARE_DIR contains the actual igblastn and igblastp binaries and also the
 # required data files. Wrappers will be installed into $PREFIX/bin that set
 # $IGDATA to point to those data files.
 mkdir -p $SHARE_DIR/bin
 
-# Copy binaries and wrappers
-for name in igblastn igblastp; do
-  mv bin/$name $SHARE_DIR/bin/
-  sed "s/igblastn/$name/g" $RECIPE_DIR/igblastn.sh > $PREFIX/bin/$name
-  chmod +x $PREFIX/bin/$name
-done
+if [[ $(uname) == Linux ]]; then
+    export CFLAGS="$CFLAGS -O2"
+    export CXXFLAGS="$CXXFLAGS -O2"
+    export CPPFLAGS="$CPPFLAGS -I$PREFIX/include -I$PREFIX/include/ncbi-vdb"
+    export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
+    export CC_FOR_BUILD=$CC
+    export AR="$AR rcs"
 
-# No wrapper needed
-mv bin/makeblastdb $PREFIX/bin/
-
-wget $IGBLAST_ADDRESS/edit_imgt_file.pl
-# Replace the hardcoded perl shebang pointing to /opt with `#!/usr/bin/env perl`.
-sed -i.backup '1 s_^.*$_#!/usr/bin/env perl_' edit_imgt_file.pl
-chmod +x edit_imgt_file.pl
-mv edit_imgt_file.pl $PREFIX/bin/
+    cd c++
 
+    # IgBLAST is based on the BLAST source code and building it produces
+    # a similar set of shared libraries if --with-dll is used. To avoid
+    # conflicts when installing IgBLAST and BLAST simultaneously,
+    # we link IgBLAST statically.
+    ./configure.orig \
+        --with-static-exe \
+        --with-mt \
+        --with-openmp \
+        --without-autodep \
+        --without-makefile-auto-update \
+        --with-flat-makefile \
+        --with-caution \
+        --without-lzo \
+        --without-debug \
+        --with-strip \
+        --with-z=$PREFIX \
+        --with-bz2=$PREFIX \
+        --with-vdb=$PREFIX \
+        --without-krb5 \
+        --without-openssl \
+        --without-gnutls \
+        --without-gcrypt \
+        --with-build-root=ReleaseMT \
+        --prefix=$PREFIX
+    make -j2
+    # Move one up so it looks like the binary release
+    mv ReleaseMT/bin .
+    mv src/app/igblast/{internal_data,optional_file} $SHARE_DIR
+else
+    # On macOS, prebuilt binaries are used
+    mv internal_data optional_file $SHARE_DIR
+fi
+mv bin/makeblastdb $PREFIX/bin/
+mv bin/{igblastn,igblastp} $SHARE_DIR/bin/
 
-# Download data files necessary to run IgBLAST. These are not included in the
-# source or binary distributions.
-# See the [IgBLAST README](ftp://ftp.ncbi.nih.gov/blast/executables/igblast/release/README)
+# Replace the shebang
+sed '1 s_^.*$_#!/usr/bin/env perl_' bin/edit_imgt_file.pl > $PREFIX/bin/edit_imgt_file.pl
+chmod +x $PREFIX/bin/edit_imgt_file.pl
 
-for IGBLAST_DIR in internal_data optional_file; do
-    mkdir -p $SHARE_DIR/$IGBLAST_DIR
-    wget -nv -r -nH --cut-dirs=5 -X Entries,Repository,Root,CVS -P $SHARE_DIR/$IGBLAST_DIR $IGBLAST_ADDRESS/$IGBLAST_DIR
+# Install wrappers
+for name in igblastn igblastp; do
+  sed "s/igblastn/$name/g" $RECIPE_DIR/igblastn.sh > $PREFIX/bin/$name
```

Comment:
```suggestion
  cat >"${PREFIX}/bin/${name}" <<EOF
#!/usr/bin/env bash
IGDATA="\${IGDATA-"${PREFIX}/share/igblast"}" exec "${PREFIX}/share/igblast/${name}" "\${@}"
EOF
```
This would be a little easier and you could remove `recipes/igblast/igblastn.sh` then.
(`conda-build` will replace the expanded `${PREFIX}` string with a placeholder that `conda` will then replace with the final environment path.)

(Ideally, we could patch the sources right away so the "`IGDATA` defaulting to `${PREFIX}/share/igblast`" would be baked in so we wouldn't need those wrappers -- but it's also fine as is now.)

Reply:
Cool, nice improvement!

Regarding patching the sources: I had a look now and probably have an idea of where to patch, but with each round of testing taking 60 minutes, I’ll skip this for now.

Reply:
> but with each round of testing taking 60 minutes, I’ll skip this for now.

Hehe, won't blame you on that one -- those were my thoughts exactly ;).
Thanks for having taken a look anyway!